### PR TITLE
F/flax split head dim

### PR DESCRIPTION
**Pull request by entrpn**
# What does this PR do?

@patrickvonplaten Adds optional flag to split_head_dim in flax attention which speeds up the diffusion process for Stable Diffusion 2.1 and SDXL.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [X] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/models/attention_flax.py [hunk 1]
```diff
@@ -131,6 +131,8 @@ class FlaxAttention(nn.Module):
             Dropout rate
         use_memory_efficient_attention (`bool`, *optional*, defaults to `False`):
             enable memory efficient attention https://arxiv.org/abs/2112.05682
+        split_head_dim (`bool`, *optional*, defaults to `False`):
+            speeds up Stable Diffusion 2.1 and XL
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention_flax.py:135]:
```suggestion
            Whether to split the head dimension into a new axis for the self-attention computation. In most cases, enabling this flag should speed up the computation for Stable Diffusion 2.x and Stable Diffusion XL.
```

**Review by patrickvonplaten — APPROVED** [review]
Very cool! @pcuenca what do you think?

**Review by pcuenca — APPROVED** [review]
Looks great, thanks a lot!

Running tests to see if anything important comes up :)

**Comment by pcuenca** [comment]
@entrpn could you please run `make style` in your branch? Otherwise I can do it later.

**Comment by entrpn** [comment]
@pcuenca I ran it already. I just re-ran it after Patrick's changes and there were no changes.

**Comment by pcuenca** [comment]
Ok, I'll merge then and we can verify later.
